Integration test failure: deployment/etcd

[test artifacts](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/24804/kubernetes-pull-test-unit-integration/23757/)

>
FAIL	k8s.io/kubernetes/pkg/registry/deployment/etcd	15.668s

```
=== RUN   TestCreate
E0502 12:43:47.954555   32628 utils.go:192] Error when getting etcd cluster members
--- FAIL: TestCreate (12.32s)
	utils.go:245: Unexpected error in waitUntilUp (timeout on waiting for etcd cluster)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x87b381]

goroutine 117 [running]:
panic(0x1d3afa0, 0xc8200120c0)
	/usr/local/go/src/runtime/panic.go:464 +0x3ff
testing.tRunner.func1(0xc8200daa20)
	/usr/local/go/src/testing/testing.go:467 +0x22a
panic(0x1d3afa0, 0xc8200120c0)
	/usr/local/go/src/runtime/panic.go:426 +0x521
k8s.io/kubernetes/pkg/registry/registrytest.NewEtcdStorage(0xc8200daa20, 0x1f8e450, 0xa, 0x0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/registrytest/etcd.go:41 +0x211
k8s.io/kubernetes/pkg/registry/deployment/etcd.newStorage(0xc8200daa20, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/deployment/etcd/etcd_test.go:41 +0x56
k8s.io/kubernetes/pkg/registry/deployment/etcd.TestCreate(0xc8200daa20)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/registry/deployment/etcd/etcd_test.go:85 +0x44
testing.tRunner(0xc8200daa20, 0x2bc9cc0)
	/usr/local/go/src/testing/testing.go:473 +0xdd
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0xae3
FAIL	k8s.io/kubernetes/pkg/registry/deployment/etcd	15.668s
```

--- a/pkg/storage/etcd/testing/utils.go
+++ b/pkg/storage/etcd/testing/utils.go
@@ -239,13 +239,13 @@ func NewEtcdTestClientServer(t *testing.T) *EtcdTestServer {
 	}
 	server.Client, err = etcd.New(cfg)
 	if err != nil {
-		t.Errorf("Unexpected error in NewEtcdTestClientServer (%v)", err)
 		server.Terminate(t)
+		t.Fatalf("Unexpected error in NewEtcdTestClientServer (%v)", err)
 		return nil
 	}
 	if err := server.waitUntilUp(); err != nil {
-		t.Errorf("Unexpected error in waitUntilUp (%v)", err)
 		server.Terminate(t)
+		t.Fatalf("Unexpected error in waitUntilUp (%v)", err)
 		return nil
 	}
 	return server

--- a/pkg/storage/etcd/testing/utils.go
+++ b/pkg/storage/etcd/testing/utils.go
@@ -27,6 +27,8 @@ import (
 	"testing"
 	"time"
 
+	"k8s.io/kubernetes/pkg/util/wait"
+
 	etcd "github.com/coreos/etcd/client"
 	"github.com/coreos/etcd/etcdserver"
 	"github.com/coreos/etcd/etcdserver/api/v2http"
@@ -186,7 +188,7 @@ func (m *EtcdTestServer) launch(t *testing.T) error {
 // waitForEtcd wait until etcd is propagated correctly
 func (m *EtcdTestServer) waitUntilUp() error {
 	membersAPI := etcd.NewMembersAPI(m.Client)
-	for start := time.Now(); time.Since(start) < 5*time.Second; time.Sleep(10 * time.Millisecond) {
+	for start := time.Now(); time.Since(start) < wait.ForeverTestTimeout; time.Sleep(10 * time.Millisecond) {
 		members, err := membersAPI.List(context.TODO())
 		if err != nil {
 			glog.Errorf("Error when getting etcd cluster members")

